src/notaryUtils.js

```diff
@@ -7,16 +7,24 @@ const getTransferState = require('./transferUtils').getTransferState
 const Payments = require('./payments')
 
 /**
+ * @returns {String} New case ID
+ */
+function createCaseID () {
+  return uuid()
+}
+
+/**
  * @param {Object} params
  * @param {URI} params.notary
  * @param {Condition} params.receiptCondition
  * @param {[Payment]} params.payments
  * @param {String} params.expiresAt
+ * @param {String} params.caseID
  * @returns {Promise<URI>} Case ID
  */
 function setupCase (params) {
   return co(function * () {
-    const caseID = params.notary + '/cases/' + uuid()
+    const caseID = params.notary + '/cases/' + (params.caseID || uuid())
```

Comment:
@lumberj the notary db only handles 40 characters but the controller handles the validation: https://github.com/interledger/five-bells-notary/blob/master/src/controllers/cases.js#L22